Initial SUSE Multilinux Manager chapter (#462)
suse-edge/suse-edge.github.io: asciidoc/components/suse-multilinux-manager.adoc

> == SUSE Manager

Reviewer: You'll also need to add the front-matter with an anchor then link to it from edge.adoc

> == SUSE Multi-Linux Manager: Keeping SUSE Linux Micro up to date

Reviewer 2: For this to show up correctly in the side menu we need to adjust like:

```
[#components-suma]
= SUSE Multi-Linux Manager
```

Then make corresponding adjustments to the following sections - see [my branch](https://github.com/hardys/suse-edge.github.io/commit/ca007d6288fee27999ff4ade11f6afcbf02b3cc7) for an example, and also the required addition to `edge-book/edge.adoc`

What is SUSE Manager.
Set up the server
Configure for SL Micro 6.0
Configure your installation image with EIB

== Keeping SLE Micro up to date

> TBD: Do we want to give customers options (Elemental all-in, direct registration against SCC), or is SUSE Manager the only fully supported (and documented) option (in SUSE Edge 3.1)?

Reviewer 2: We already document SCC registration so I think it makes sense to describe SUSE Manager as an optional alternative?

Author: For now, I'll just not mention the other alternatives here. But we might need an intro snippet somewhere in the docs to list all the available OS patching options and explain which is best when.

=== Deploy SUSE Multi-Linux Manager Server

If you already have an instance of the latest version of SUSE Multi-Linux Manager 5.0 running, you can skip this step.

You can run SUSE Multi-Linux Manager Server on a dedicated physical server, as a virtual machine on your own hardware, or in the cloud. Pre-configured virtual machine images for SUSE Multi-Linux Server are provided for supported public clouds.

In this quick start we're using the "qcow2" image "SUSE-Manager-Server.x86_64-5.0.2-Qcow-2024.12.qcow2" for x86-64 that you can find at https://www.suse.com/download/suse-manager/ or in the SUSE Customer Center. This image will work as a virtual machine on hypervisors like KVM. Please always check for the newest version of the image and use it for new installations.

You can also install SUSE Multi-Linux Manager Server on any of the other supported hardware architectures.

Once you have downloaded the image, create a virtual machine that meets at least the following minimal hardware specifications:

* 16 GB RAM
* 4 physical or virtual cores

With the qcow2 image, there is no need to install the operating system. You can directly attach the image as your root partition.

> You will also need an additional block device to store the SUSE Multi-Linux Manager database and all RPM packages and files SUSE Multi-Linux Manager needs to operate. With your VM management tool, create a block device that has at least 100 GB.

Reviewer 3: Might I suggest that we move the block storage requirements to the list above so we're exhaustive in the list rather than adding this sentence afterwards?

Author: Done. Also added a snippet for using the mgr-storage-server command for actually configuring the storage for Multi-Linux Manager.

> You need to set up the network so that your edge nodes can later access SUSE Multi-Linux Manager Server with a fully qualified host name!

Reviewer 3: ```suggestion
You need to set up the network so that your edge nodes can later access SUSE Multi-Linux Manager Server with a fully qualified domain name (FQDN).
```

Author: Done.

Once you have downloaded the image, create a virtual machine that meets at least the following minimal hardware specifications:

* 16 GB RAM
* 4 physical or virtual cores

With the qcow2 image, there is no need to install the operating system. You can directly attach the image as your root partition.

You will also need an additional block device to store the SUSE Manager database and all RPM packages and files SUSE Manager needs to operate. With your VM management tool, create a block device that has at least 100 GB.

You need to set up the network so that your edge nodes can later access SUSE Manager Server with a fully qualified host name!

When you boot SUSE Manager for the first time you need to perform some initial configuration:

* Select your keyboard layout
* Accept the license agreement
* Select your time zone
* Enter the root password for the operating system

The next steps need to be done as the "root" user:

FIXME: DO WE NEED TWO CODES? WHAT EXACTLY DOES SOMEONE WITH AN EDGE SUBSCRIPTION GET?

For the next step you need two registration codes that you can find in the SUSE Customer Center:

> * Your registration code for SLE Micro 5.5

Reviewer 2: Should be 6.0

Author: Correct. I still need to verify which code to use when the Edge Suite was purchased.

A typical advanced use case for activation keys would be to assign your test clusters to the software channels with the latest updates and your production clusters to software channels that only get those latest updates once you've tested them in the test cluster.

Click "+ Create Key"
Choose a short description, e.g., "Edge Nodes".
Provide a unique name that identifies the key, e.g., "edge-x86_64" for your edge nodes with x86_64 hardware architecture.
A number prefix is automatically added to the key. For the default organization, the number is always "1". If you create additional organizations in SUSE Manager and create keys for them, that number may differ.

If you haven't created any cloned software channels, you can keep the setting for the Base Channel to "SUSE Manager Default". This will automatically assign the correct SUSE update repository for your edge nodes.

> As "Child Channel", select the "include recomended" slider for the hardware architecture your activation key is used for. This will add the "SUSE-Manager-Tools-For-SL-Micro-6.0" channel.

Reviewer 4: ```suggestion
As "Child Channel", select the "include recommended" slider for the hardware architecture your activation key is used for. This will add the "SUSE-Manager-Tools-For-SL-Micro-6.0" channel.
```

Author: Fixed.

> For a minimal lab setup, we can actually use the same virtual machine SUSE Multi-Linux Manager Server is running on. Please make sure that you have enough disk space in the virtual machine! This is not a recommended setup for production use. See https://documentation.suse.com/suse-edge/3.2/html/edge/quickstart-eib.html#id-prerequisites-2[here] for host operating systems we have tested Edge Image builder with.

Reviewer 2: We should use an internal reference e.g `<<id-prerequisites-2>>` (again see my branch linked above)

On the "Groups" tab, add the group you've created before. All nodes that are onboarded using this activation key will automatically added to that group.

=== Create a customized installation image with Edge Image Builder

To use Edge Image Builder, you only need an environment where you can start a Linux-based container with podman.

For a minimal lab setup, we can actually use the same virtual machine SUSE Manager Server is running on. Please make sure that you have enough disk space in the virtual machine! This is not a recommended setup for production use. See ... for host operating systems we have tested Edge Image builder with.

Log into your SUSE Manager Server host as root.

Pull the Edge Image Builder container:

> podman pull registry.opensuse.org/isv/suse/edge/edgeimagebuilder/containerfile-sp6/suse/edge-image-builder:1.1.0.rc2

Reviewer 2: Should be `registry.suse.com/edge/3.1/edge-image-builder:1.1.0`

Author: Fixed, using the variables like in the EIB Quick Start.

> podman pull registry.suse.com/edge/{version-edge-registry}/edge-image-builder:{version-eib}

Reviewer 2: Note your branch is outdated, so will need a rebase for it to build with these attribute references.

> In this quickstart we're using the "self-install" flavor of the SUSE Linux Micro image. That image can later be written to a physical USB thumb drive that you can use to install on physical servers. If your server has the option of remote-attaching installation ISOs via a BMC (Baseboard Management Controller), you can also use that approach. Finally that image can also be used with most virtualization tools.

Reviewer 3: ```suggestion
In this quickstart we're using the "self-install" flavor of the SUSE Linux Micro image. That image can later be written to a physical USB thumb drive that you can use to install on physical servers. If your server has the option of remote attachment of installation ISOs via a BMC (Baseboard Management Controller), you can also use that approach. Finally that image can also be used with most virtualization tools.
```

Author: Done.

FIXME: Final version of the container!

Create the directory "/opt/eib" and a sub-directory "base-images":

mkdir -p /opt/eib/base-images

FIXME: Create an example on github and link it.

In this quickstart we're using the "self-install" flavor of the SLE Micro image. That image can latter be written to a physical USB thumb drive that you can use to install on physical servers. If your server has the option of remote-attaching installation ISOs via a BMC (Baseboard Management Controller), you can also use that approach. Finally that image can also be used with most virtualization tools.

If you either want to preload the image directly to a physical node or directly start it from a VM, you can also use the "raw" image flavor.

Download or copy the image

> SL-Micro.x86_64-6.0-Default-SelfInstall-GM.install.iso to the "base-images" directoy and name it "slemicro.iso".

Reviewer 2: Should be the new GM2 image

Author: Done.

> `SL-Micro.x86_64-6.0-Default-SelfInstall-GM2.install.iso` to the `base-images` directoy and name it "slemicro.iso".

Reviewer 3: You don't actually need to rename it. I think this is an unnecessary step and perhaps adds confusion. I'd suggest we just use the full name in the configuration file.

Author: I've kept the rename step for now. It has the benefit of not having to always touch the config when you download a new version. At least when doing rapid iterations (and not using templated YAML files), this, IMHO, makes sense.

apiVersion: 1.0
image:
  imageType: iso
  arch: x64_64
  baseImage: slemicro.iso
  outputImageName: eib-image.iso
operatingSystem:
  users:
  - username: root
    createHomeDir: true
    encryptedPassword: $6$aaBTHyqDRUMY1HAp$pmBY7.qLtoVlCGj32XR/Ogei4cngc3f4OX7fwBD/gw7HWyuNBOKYbBWnJ4pvrYwH2WUtJLKMbinVtBhMDHQIY0
  keymap: de
  systemd:
    enable:
      - cockpit.socket
  packages:
    noGPGCheck: true
  suma:

>     host: ${fully qualified hostname of your SUSE Multi-Linux Manager Server}

Author: So, I've double-checked, and the correct term is indeed "hostname" (once it contains the actual host's unique name, not just the domain/subdomain parts). The FQDN is just the domain part of it. So, where it made sense, I've elaborated on this, mentioning the "hostname" needs a FQDN part.

    activationKey: 1-edge-x86_64
----

> Edge Image Builder can also configure the network, automatically install Kubernetes on the node, and even deploy applications via helm charts. See [here] for more comprehensive examples.

Reviewer 3: Where is [here]? We should update this link.

For `baseImage`, specify the actual name of the ISO in the `base-images` directory that you want to use.

> In this example, the root password would be "root". See https://documentation.suse.com/suse-edge/3.2/html/edge/quickstart-eib.html#id-configuring-os-users[here] for creating password hashes for the secure password you want to use.

Reviewer 3: We should templatise the Edge version here too.

Author: Done.

Set the keymap to the actual keyboard layout you want the system to have after installation.

> We use the option `noGPGCheck: true` because we aren't going to provide a GPG key to check RPM packages. See [here] for a more secure setup that we recommend for production use.

Reviewer 3: Another [here] to update

Author: Done.

As mentioned several times, your SUSE Multi-Linux Manager host requires a fully qualified hostname that can be resolved in the network your edge nodes will boot into.

The value for `activationKey` needs to match the key you've created in SUSE Multi-Linux Manager. 

To build an installation image that automatically registers your edge nodes to SUSE Multi-Linux Manager after installation, you also need to prepare two artifacts:

* the Salt minion package that installs the management agent for SUSE Multi-Linux Manager
* the CA certificate of your SUSE Multi-Linux Manager server

==== Download the venv-salt-minion package

In `/opt/eib`, create a subdirectory `rpms`.

Download the package `venv-salt-minion` from your SUSE Multi-Linux Manager server into that directory. You can either find it via the web UI by finding the package under `Software > Channel List` and download it from the SUSE-Manager-Tools ... channel or download it from the SUSE Multi-Linux Manager "bootstrap repo" with a tool like curl:

[,shell]
----

> curl http://${HOSTNAME_OF_SUSE_MANAGER}/pub/repositories/slmicro/6/0/bootstrap/x86_64/venv-salt-minion-3006.0-1.1.x64_64.rpm

Author: Done.

In "/opt/eib", create a subdirectory "rpms".

Download the package "venv-salt-minion" from your SUSE Manager server into that directory. You can either find it via the web UI by finding the package under "Software > Channel List" and download it from the SUSE-Manager-Tools ... channel or download it from the SUSE Manager "bootstrap repo" with a tool like curl:

curl http://${HOSTNAME_OF_SUSE_MANAGER}/pub/repositories/slmicro/6/0/bootstrap/x86_64/venv-salt-minion-3006.0-1.1.x64_64.rpm

FIXME: Exact curl command

=== Download the SUSE Manager CA certificate

In "/opt/eib", create a subdirectory "certificates"

Download the CA certificate from SUSE Manager into that directory:

curl http://${HOSTNAME_OF_SUSE_MANAGER}/pub/RHN-ORG-TRUSTED-SSL-CERT

You need to rename it to "RHN-ORG-TRUSTED-SSL-CERT.crt". Edge Image Builder will make sure that the certificate is installed and activated on the edge node during installation.

Now you can run Edge Image Builder:

cd /opt/eib

> podman run --rm -it --privileged -v ./:/eib registry.opensuse.org/isv/suse/edge/edgeimagebuilder/containerfile-sp6/suse/edge-image-builder:1.1.0.rc2 build --definition-file iso-definition.yaml

Reviewer 2: Same as above should be the registry.suse.com version

Author: Done.

> You need to rename it to `RHN-ORG-TRUSTED-SSL-CERT.crt`. Edge Image Builder will make sure that the certificate is installed and activated on the edge node during installation.

Reviewer 3: Suggest we add a "WARN" block for this, as if the user doesn't have the `.crt` extension, EIB will ignore it, and nodes will fail to register to systems with self-signed certificates.

Author: Done.
